src/tools/compiletest/src/runtest.rs

```diff
@@ -661,6 +669,19 @@ impl<'test> TestCx<'test> {
     }
 
     fn run_debuginfo_cdb_test_no_opt(&self) {
+        let exe_file = self.make_exe_name();
+
+        // Existing PDB files are update in-place. When changing the debuginfo
+        // the compiler generates for something, this can lead to the situation
+        // where both the old and the new version of the debuginfo for the same
+        // type is present in the PDB, which is very confusing.
+        // Therefore we delete any existing PDB file before compiling the test
+        // case.
```

Comment:
This is fine but I believe this happens because `link.exe` runs in incremental mode by default (https://docs.microsoft.com/en-us/cpp/build/reference/incremental-link-incrementally?view=msvc-170). Passing `/INCREMENTAL:NO` for these tests in the test runner could be another solution.